packages/flutter_tools/test/general.shard/android/android_sdk_test.dart

```diff
@@ -342,6 +342,127 @@ void main() {
       Config: () => config,
     });
   });
+
+  const Map<String, String> llvmHostDirectoryName = <String, String>{
+    'macos': 'darwin-x86_64',
+    'linux': 'linux-x86_64',
+    'windows': 'windows-x86_64',
+  };
+
+  for (final String operatingSystem in <String>['windows', 'linux', 'macos']) {
+    final FileSystem fileSystem;
+    final String extension;
+    if (operatingSystem == 'windows') {
+      fileSystem = MemoryFileSystem.test(style: FileSystemStyle.windows);
+      extension = '.exe';
+    } else {
+      fileSystem = MemoryFileSystem.test();
+      extension = '';
+    }
+    testUsingContext('ndk executables $operatingSystem', () {
```

Comment:
Can this be `testWithoutContext`? Same question for all tests in this patch.

Reply:
No, it uses `overrides:` of the `Platform` to test all platforms on any host.

For almost any test in this file: the `fileSystem` is overriden with a memory file system.

Reply:
testWithoutContext only works if none of the code called tries to grab any globals from the zone context. It looks like `AndroidSdk.locateAndroidSdk()` does this a bunch. However, it's not clear to me that we're necessarily interested in testing that function there, and could directly construct an `AndroidSdk()` instance, passing `sdkDir` (I think we'd also need to pass the FakePlatform to `createSdkDirectory`